component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -299,58 +300,111 @@ public String getContextIdentifier(HttpServletRequest httpServletRequest) {
-                headerString = headerString.trim().replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg",
-                        smsMessage + otpToken);
-                headerList = headerString.split(",");
-                for (String aHeaderList : headerList) {
-                    String[] header = aHeaderList.split(":");
-                    connection.setRequestProperty(header[0], header[1]);
-                }
-            }
-            if (httpMethod.toUpperCase().equals(SMSOTPConstants.GET_METHOD)) {
-                connection.setRequestMethod(SMSOTPConstants.GET_METHOD);
-            } else if (httpMethod.toUpperCase().equals(SMSOTPConstants.POST_METHOD)) {
-                connection.setRequestMethod(SMSOTPConstants.POST_METHOD);
-                if (!payload.isEmpty()) {
-                    payload = payload.replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg", smsMessage + otpToken);
+            String subUrl = smsProviderUrl.getProtocol();
+            if (subUrl.equals("https")) {
+                httpsConnection = (HttpsURLConnection) smsProviderUrl.openConnection();
+                httpsConnection.setDoInput(true);
+                httpsConnection.setDoOutput(true);
+                String[] headerList;
+                if (!headerString.isEmpty()) {
+                    headerString = headerString.trim().replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg",
+                            smsMessage + otpToken);
+                    headerList = headerString.split(",");
+                    for (String aHeaderList : headerList) {
+                        String[] header = aHeaderList.split(":");
+                        httpsConnection.setRequestProperty(header[0], header[1]);
+                    }
                 }
-                OutputStreamWriter writer = null;
-                try {
-                    writer = new OutputStreamWriter(connection.getOutputStream(), "UTF-8");
-                    writer.write(payload);
-                } catch (IOException e) {
-                    throw new AuthenticationFailedException("Error while posting payload message", e);
-                } finally {
-                    if (writer != null) {
-                        writer.close();
+                if (httpMethod.toUpperCase().equals(SMSOTPConstants.GET_METHOD)) {
+                    httpsConnection.setRequestMethod(SMSOTPConstants.GET_METHOD);
+                } else if (httpMethod.toUpperCase().equals(SMSOTPConstants.POST_METHOD)) {
+                    httpsConnection.setRequestMethod(SMSOTPConstants.POST_METHOD);
+                    if (!payload.isEmpty()) {
+                        payload = payload.replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg", smsMessage + otpToken);
+                    }
+                    OutputStreamWriter writer = null;
+                    try {
+                        writer = new OutputStreamWriter(httpsConnection.getOutputStream(), "UTF-8");
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/8